internal/querycoordv2/meta/resource_group.go

```diff
@@ -201,6 +252,36 @@ func (rg *ResourceGroup) MeetRequirement() error {
 			rg.cfg.Requests.NodeNum,
 		)
 	}
+
+	// check rg node filter
+	if rg.cfg.NodeFilter != nil {
+		nodeLabelSet := typeutil.NewSet(rg.GetConfig().GetNodeFilter().GetPreferNodeLabels()...)
+		nodeFilter := func(nodeID int64) bool {
+			nodeInfo := nodeMgr.Get(nodeID)
+			if nodeInfo == nil {
+				return false
+			}
+			return nodeLabelSet.Contain(nodeInfo.Label())
```

Comment:
In this implementation, a node without any label will always be treated as a dirty node if the resource group has label filter. It's wired for a filter whose function is supplying a prefer semantic.

And what is the behavior when a dirty node is found?
It seems that current recovery operation cannot fix the node assignments into the best one if a resource group's filter field is changed.

Reply:
1. the expected assignment means that all node in resource group match the label filter, and other nodes will be regard as dirty node.
2. if dirty node exists, resource group should return `false` in `MeetRequirement` to trigger auto recover.
3. `resource_manager` will try to assign available prefer node to resource group in `recoverMissingNodeRG` to meet resource group's requirement

Reply:
I'm considering the following state:

- Resource Group `A` has a dirty node `a` with config `require 1, limit 1`;
- Resource group `B` want `a`, has a node `c`, with config `require 1, limit 1`. 

How does `A` exchange the node with `B` in current implementation?